### PR TITLE
Enable testserver to be able to handle multiple stores.

### PR DESCRIPTION
**Pull request by BramGruneir**
I'll be adding a test with multiple stores in another pr.

### CODE REVIEW

File: server/testserver.go [hunk 1]
```diff
@@ -120,15 +120,32 @@ func (ts *TestServer) Start() error {
...
-	if ts.Engine == nil {
-		ts.Engine = engine.NewInMem(proto.Attributes{}, 100<<20)
+	// Ensure we have the correct number of engines. Add in in-memory ones where
+	// needed.  There must be at least one stone (and hence engine).
+	if ts.StoresPerNode < 1 {
+		ts.StoresPerNode = 1
 	}
-	ts.Ctx.Engines = []engine.Engine{ts.Engine}
+	if ts.Engines == nil {
+		ts.Engines = make([]engine.Engine, ts.StoresPerNode)
```
Comment (bdarnell) [inline server/testserver.go]: This is unnecessary; the append on line 134 will take care of it. In general I would only use `make()` in cases where it's likely to make a performance difference.

File: server/testserver.go [hunk 2]
```diff
@@ -120,15 +120,32 @@ func (ts *TestServer) Start() error {
...
+	}
+	ts.Ctx.Engines = make([]engine.Engine, ts.StoresPerNode)
```
Comment (bdarnell) [inline server/testserver.go]: It's weird to me that we write the engines back out to the context, but assuming that's something we need here why not just set `ts.Ctx.Engines = ts.Engines` after creating the engines (or use `append(nil, ts.Engines)` to make a copy)

**Comment by bdarnell** [comment]
LGTM

File: server/testserver.go (hunk 2 above)
Comment (BramGruneir) [inline server/testserver.go]: Fixed.  I just set it directly for now.

File: server/testserver.go (hunk 1 above)
Comment (BramGruneir) [inline server/testserver.go]: Fixed.

**Comment by mrtracy** [comment]
LGTM

**Pull request merged**